### PR TITLE
Fix parse-data command, fix API bug, and general upkeep

### PR DESCRIPTION
**Pull request by michplunkett**
## What does this pull request add?
Fix the `parse-data` flag, the API bug, add state files so that we can run tests, and update `Author(s)` statements to the current ED standard, and move the `to_json` function to the `utils` package.

## Are there any technical things that should be noted for this PR?
Nay.

## How was it tested?
- [x] Previously working test cases are still passing.

<details><summary>Output</summary>

```
(reproductive-rights-data-project-py3.11) michaelp@MacBook-Air-3 reproductive-rights-data-project % make test
pytest -vs ./test/
============================================================================ test session starts =============================================================================
platform darwin -- Python 3.11.1, pytest-7.2.1, pluggy-1.0.0 -- /Users/michaelp/Documents/GitHub/reproductive-rights-data-project/.venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/michaelp/Documents/GitHub/reproductive-rights-data-project
collected 11 items                                                                                                                                                           

test/test_api.py::test_get_api_data PASSED
test/test_api.py::test_set_default_types PASSED
test/test_api.py::test_fill_in_missing_data PASSED
test/test_api.py::test_add_missing_states PASSED
test/test_data_handling.py::test_make_row_dicts PASSED
test/test_data_handling.py::test_set_default_types PASSED
test/test_data_handling.py::test_clean_ansirh PASSED
test/test_data_handling.py::test_translate_code_to_state PASSED
test/test_data_handling.py::test_split_by_state PASSED
test/test_data_handling.py::test_split_by_zip PASSED
test/test_util.py::test_to_json Writing out to json file
PASSED

============================================================================= 11 passed in 0.32s =============================================================================
(reproductive-rights-data-project-py3.11) michaelp@MacBook-Air-3 reproductive-rights-data-project % 

```
</details>

- [x] I ran `pylint` and have made the bulk of changes that it has requested.
- [x] I ran the `black` formatter and pushed the changes.
- [x] Ran `make api` command.

<details><summary>Output</summary>

```
(reproductive-rights-data-project-py3.11) michaelp@MacBook-Air-3 reproductive-rights-data-project % make api
python ./ --api
Writing out to json file
(reproductive-rights-data-project-py3.11) michaelp@MacBook-Air-3 reproductive-rights-data-project % 
```
</details>

- [x] Ran `make parse-data` command.

<details><summary>Output</summary>

```
(reproductive-rights-data-project-py3.11) michaelp@MacBook-Air-3 reproductive-rights-data-project % make parse-data
python ./ --parse-data
(reproductive-rights-data-project-py3.11) michaelp@MacBook-Air-3 reproductive-rights-data-project % 
```
</details>

### CODE REVIEW

File: api/service.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 This file works as the central point for running the api package.
 """
 
-from abortion_policy_api import get_and_save_abortion_policy_api_data
+from api.abortion_policy_api import get_and_save_abortion_policy_api_data
```
Comment (michplunkett) [inline api/service.py:5]: Problem with `parse-data` command.

File: .gitignore [hunk 2]
```diff
@@ -3,7 +3,10 @@
 .vscode
 
 # Stored data
-data/
+data/*
+!data/states.txt
+!data/state_abbreviations.csv
+!data/zip_code_database.csv
```
Comment (michplunkett) [inline .gitignore:9]: Exceptions for state files, as they'll be needed for tests, etc.

File: api/abortion_policy_api.py [hunk 3]
```diff
@@ -194,4 +194,4 @@ def to_json(data, file_names):
 
     for i, file_name in enumerate(file_names):
         with open(file_name, "w", encoding=STANDARD_ENCODING) as f:
-            json.dump(data[list(data.keys())[i]], f, indent=1)
+            json.dump(data[i], f, indent=1)
```
Comment (michplunkett) [inline api/abortion_policy_api.py:197]: Here's the bug fix.

File: test/test_api.py [hunk 4]
```diff
@@ -148,7 +148,10 @@ def test_get_api_data():
 
 
 def test_to_json():
-    policy_areas = {"gestational": json.loads(gestational_data)}
+    """
+    Author(s): Chanteria Milner, Michael Plunkett
+    """
+    policy_areas = [json.loads(gestational_data)]
```
Comment (michplunkett) [inline test/test_api.py:154]: Here's the fix that was needed to update the test.

File: api/abortion_policy_api.py [hunk 5]
```diff
@@ -186,6 +186,8 @@ def to_json(data, file_names):
         data (dict): list of dictionaries to output to JSON
         file_names (list): list of file names to export to
     """
+    # TODO: Merge with `to_json` function in the data_handling function
+    # TODO: and put it in the utils package.
```
Comment (michplunkett) [inline api/abortion_policy_api.py:190]: @ehabich or @chanteriam , could y'all take a stab at this?

File: test/test_api.py [hunk 6]
```diff
@@ -159,6 +162,9 @@ def test_to_json():
 
 
 def test_set_default_types():
+    """
+    Author(s): Chanteria Milner
+    """
```
Comment (michplunkett) [inline test/test_api.py:167]: Update on the `Author(s)` statements for credit.

**Review by necabotheking — APPROVED** [review]

**Pull request merged**